### PR TITLE
Made library work for mojo 24.3

### PR DESCRIPTION
**Pull request by andresnowak**
Updated library for mojo 24.3

### CODE REVIEW

**Comment by Benny-Nottonson** [comment]
Since this will create a merge conflict with #63, might be better to merge that first then resolve conflicts

**Comment by andresnowak** [comment]
Oh yeah thats why i made it a draft, i just created it so people know it exists.

File: basalt/utils/collection.mojo [hunk 1]
```diff
@@ -41,20 +41,20 @@ struct Collection(CollectionElement, Sized):
...
+            initialize_pointee_move((self.data + i), (existing.data + i)[])
 
     @always_inline("nodebug")
     fn __del__(owned self):
         """
         Destructor for the Collection.
         """
         for i in range(self.size):
-            _ = (self.data + i).take_value()
+            _ = move_from_pointee((self.data + i))
```
Comment (soraros) [inline basalt/utils/collection.mojo:57]: You might use `destroy_pointee` here.

**Comment by Benny-Nottonson** [comment]
You also might wanna rework this branch as not minutes after draft, Mojo 23.3 released officially 🔥

**Comment by andresnowak** [comment]
It shouldn't be necessary the changelog for 23.3 is the same as the last nightly from what I could see, even though i will test it when the other branches are merged with 23.3 but it is the same pretty sure.

File: basalt/utils/collection.mojo (hunk 1 above)
Comment (andresnowak) [reply basalt/utils/collection.mojo:57]: oh yeah maybe let me test it.

Comment (andresnowak) [reply basalt/utils/collection.mojo]: Yeah it worked correctly

**Review by StijnWoestenborghs — APPROVED** [review]

File: basalt/utils/collection.mojo [hunk 2]
```diff
@@ -119,27 +119,29 @@ struct Collection(CollectionElement, Sized):
...
         Returns a reference to the tensor with the given symbol.
         """
         var index = Reference(self)[].get_index(symbol.name)
 
-        return Reference(
-            __mlir_op.`lit.ref.from_pointer`[
-                _type = Reference[Tensor[dtype], mutability, lifetime].mlir_ref_type
-            ]((Reference(self)[].data + index).value)
-        )
+        return (Reference(self)[].data + index)[]
```
Comment (StijnWoestenborghs) [inline basalt/utils/collection.mojo:130]: This is a nice cleanup

**Pull request merged**